client_it_test.go

```diff
@@ -939,6 +939,17 @@ func TestFetchStatus(t *testing.T) {
 	}
 }
 
+func TestFetchInfo(t *testing.T) {
+	client := getClient()
+	info, err := client.Info()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if info.ShardWidth == 0 {
```

Comment:
I think adding checks to other fields too would be trivial and useful.

Reply:
It might, but that would have to wait on the server being updated. My thought was to wait a release cycle or so after the server updates; I don't want the client to require a newer version of the server if it doesn't have to.

Reply:
The tests run against pilosa/master on CI, so I think that won't be a problem.

Reply:
It would right now, because the corresponding change isn't in yet. I'm also concerned a bit about users who might want to update the client without having updated the server yet; I don't want to break the tests in the absence of an underlying compatibility problem.

Reply:
The way I deal with changes on the server which are not yet merged to master is mentioning the required server-side PR in the ticket (description or a comment) and maybe mark the issue as `[DON'T MERGE]`. We can merge the PR after the corresponding server-side PR is merged. Would that be OK?

Reply:
I think so? Mostly my vague concern is that I'd rather not introduce a test failure for people using slightly-older Pilosa. If someone's not using this feature, and doesn't care about it, it introduces a possible failure even though go-pilosa is otherwise compatible with the older server, so far as I know?

Reply:
I see your point, but making the tests pass is trivial in that case, upgrading the server is adequate.  We release a new version of Pilosa roughly each month, so if we supported 2 older Pilosa releases for **client-side tests** we would have to postpone writing tests for 2 months. That means, for two months we wouldn't know whether those new features worked or not on the client-side. As a bonus, it would be much harder to write tests for all of those features at once instead of one by one (if we ever remember writing the tests after 2 months the new features are introduced).

I think the benefits of having timely client-side tests far outweighs the benefits of  **client-side tests** being incompatible with older Pilosa servers.

Reply:
... That's a good point. I'll add some tests for the other fields. (Not for Hz, because I don't actually have confidence that it'll reliably have data, yet; it's guesswork.)